src/cosmicqc/scdataframe.py

```diff
@@ -0,0 +1,164 @@
+        """
+
+        if isinstance(data, SCDataFrame):
+            # if data is an instance of SCDataFrame, use its data_source and data
+            self.data_source = data.data_source
+            self.data = data.data
+
+        elif isinstance(data, pd.Series):
+            # if data is a pd.Series, remember this within the data_source attr
+            self.data_source = "pandas.Series"
+            # also cast the series to a dataframe
+            self.data = pd.DataFrame(data)
+
+        elif isinstance(data, pd.DataFrame):
+            # if data is a pd.DataFrame, remember this within the data_source attr
+            self.data_source = "pandas.DataFrame"
+            self.data = data
+
+        elif isinstance(data, (pathlib.Path, str)):
+            # if the data is a string, remember the original source
+            # through a data_source attr
+            self.data_source = data
+
+            # interpret the data through pathlib
+            data_path = pathlib.Path(data)
+
+            # Read the data from the file based on its extension
+            if (
+                data_path.suffix == ".csv"
+                or data_path.suffix in (".tsv", ".txt")
+                or data_path.suffixes == [".csv", ".gz"]
+            ):
+                # read as a CSV, CSV.GZ, .TSV, or .TXT file
+                self.data = pd.read_csv(data, **kwargs)
+            elif data_path.suffix == ".parquet":
+                # read as a Parquet file
+                self.data = pd.read_parquet(data, **kwargs)
+            else:
+                raise ValueError("Unsupported file format for SCDataFrame.")
+        else:
+            raise ValueError("Unsupported data type for SCDataFrame.")
+
+    def export(
+        self: SCDataFrame_type, file_path: str, **kwargs: Dict[str, Any]
+    ) -> None:
+        """
+        Exports the underlying pandas DataFrame to a file.
+
+        Args:
+            file_path (str): The path where the DataFrame should be saved.
+            **kwargs: Additional keyword arguments to pass to the pandas to_* methods.
+        """
+
+        data_path = pathlib.Path(file_path)
+
+        # export to csv
+        if ".csv" in data_path.suffixes:
+            self.data.to_csv(file_path, **kwargs)
+        # export to tsv
+        elif any(elem in data_path.suffixes for elem in (".tsv", ".txt")):
+            self.data.to_csv(file_path, sep="\t", **kwargs)
+        # export to parquet
+        elif data_path.suffix == ".parquet":
+            self.data.to_parquet(file_path, **kwargs)
+        else:
+            raise ValueError("Unsupported file format for export.")
+
+    def __call__(self: SCDataFrame_type) -> pd.DataFrame:
+        """
+        Returns the underlying pandas DataFrame.
+
+        Returns:
+            pd.DataFrame: The data in a pandas DataFrame.
+        """
+        return self.data
+
+    def __repr__(self: SCDataFrame_type) -> pd.DataFrame:
```

Comment:
Wouldn't this return not be a `pd.DataFrame` exactly here, but the `repr`? Can you update this docstring (I can't remember the exact name here now) to add this?

Something like this, but this probably isn't correct or might not be necessary.

```suggestion
    def __repr__(self: SCDataFrame_type) -> repr(pd.DataFrame):
```

Reply:
Thanks, good point, yes I believe it'd be a `str` based on [Pandas `__repr__`](https://github.com/pandas-dev/pandas/blob/main/pandas/core/frame.py#L1150).